packages/@aws-cdk-testing/framework-integ/test/aws-ec2/test/integ.instance-m1-ultra.ts

```diff
@@ -0,0 +1,60 @@
+import * as cdk from 'aws-cdk-lib';
+import { IntegTest } from '@aws-cdk/integ-tests-alpha';
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+
+const app = new cdk.App();
+
+class TestStack extends cdk.Stack {
+  constructor(scope: cdk.App, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+    const az = 'us-east-1b';
+
+    const vpc = new ec2.Vpc(this, 'VPC', {
+      cidr: '10.0.0.0/16',
+      natGateways: 0,
+      availabilityZones: [az],
+      subnetConfiguration: [
+        {
+          cidrMask: 24,
+          name: 'Private',
+          subnetType: ec2.SubnetType.PRIVATE_ISOLATED,
+        },
+      ],
+    });
+
+    const privateSubnet = vpc.selectSubnets({
+      availabilityZones: [az],
+      subnetType: ec2.SubnetType.PRIVATE_ISOLATED,
+    }).subnets[0];
+
+    const instanceType = ec2.InstanceType.of(ec2.InstanceClass.MAC2_M1ULTRA, ec2.InstanceSize.METAL);
+
+    const dedicatedHost = new ec2.CfnHost(this, 'MyDedicatedHost', {
+      availabilityZone: az,
+      instanceType: instanceType.toString(),
+    });
```

Comment:
Does this integ-test even work? An EC2 Mac dedicated host cannot be deleted until 24 hours after the creation, so I guess integ-runner will fail when it tries to delete the stack.

> Mac instances are available only as bare metal instances on [Dedicated Hosts](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/dedicated-hosts-overview.html), with a minimum allocation period of 24 hours before you can release the Dedicated Host. https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-mac-instances.html

Given this restriction, I don't think it's a good idea to add an integration test for EC2 Mac. Even if we set the host's removalPolicy to retain, we have to manually remove the host 24 hours later, which is easily forgotten and not ideal. Also, it [costs](https://aws.amazon.com/jp/ec2/dedicated-hosts/pricing/) at least 120USD (5USD/h * 24h for mac2-m1ultra) every time we update the snapshot.

Since CDK currently only provides the enum for EC2 Mac instance type, I think there's little benefit to testing it with the expensive integ test. Any thoughts? @moelasmar @badmintoncryer

Reply:
@tmokmss Thank you for your comment. I completely agree with your opinion. I also feel that the benefits of conducting an integration test in this case are minimal.

Separately, as you mentioned, the integration test is failing. However, since the deployment was successful and the snapshot was generated, I was considering including it in the PR as it is.

I have encountered a similar situation before where a test failed due to resource deletion issues, but it’s good to know that this can be avoided by setting the RemovalPolicy. This is very helpful information.

Reply:
thanks @tmokmss, it is a good point. I was thinking that even it is some new enum value, but some how we need a testing coverage, so incase if it got removed by mistake some how, we still have some testing to cover us. So, I think some unit testing is enough. we can add a unit test case for it, and it will be enough.

Reply:
Thanks. Yes I agree that unit test will suffice for that purpose.